### DISCUSSION TITLE
Strawberry Shake: All auto generated classes exist in duplicate

### DISCUSSION THREAD

**Discussion by MartinKuhne** [ChilliCream/hotchocolate#5113] [Q&A]
I have an odd issue after following the getting started instructions for console applications. This is on VS 2022 .net 6.0. It looks like all the generated classes are in there twice.

![image](https://user-images.githubusercontent.com/9038432/171484566-bd1e682b-083e-4755-a9d0-62ad33649cac.png)

config

```
{
  "version": 1,
  "isRoot": true,
  "tools": {
    "strawberryshake.tools": {
      "version": "12.9.0",
      "commands": [
        "dotnet-graphql"
      ]
    }
  }
}
```

rc (all defaults other than adding the namespace)

```
{
  "schema": "schema.graphql",
  "documents": "**/*.graphql",
  "extensions": {
    "strawberryShake": {
      "name": "GraphClient",
      "namespace": "(our namespace)",
      "url": "(server url)",
      "dependencyInjection": true,
      "strictSchemaValidation": true,
      "hashAlgorithm": "md5",
      "useSingleFile": true,
      "requestStrategy": "Default",
      "outputDirectoryName": "Generated",
      "noStore": false,
      "emitGeneratedCode": true,
      "razorComponents": false,
      "records": {
        "inputs": false,
        "entities": false
      },
      "transportProfiles": [
        {
          "default": "Http",
          "subscription": "WebSocket"
        }
      ]
    }
  }
}
```

**Comment by bieri** [comment]
The only way I have been able to get around this is to set emitGeneratedCode to false.

**Comment by seangwright** [comment]
I've been fighting with the exact same thing. Pretty annoying if you don't want to have to run the source generators on _every_ build, which is what `"emitGeneratedCode": false` would require.

**Reply by bieri** [reply]
That is exactly what ends up happening unfortunately, which forced me to put all this into a separate project to isolate the graphql bits from the rest of my code, so it only builds on when that project is dirty.

**Reply by seangwright** [reply]
That's a good idea! I might try that out.

**Comment by PascalSenn** [comment]
I can highly recomment to update to version 13 of Strawberry shake. 

Because Blazor is also using source generators, we have decided to migrate away from them and go back to the good old MS Build solution. The generation of the code is now rock solid. It should now just work. There is also not really a point anymore of emitting the generated sources, as this is just a build artefact that does not belong into source control.

checkout https://chillicream.com/blog/2023/02/08/new-in-hot-chocolate-13#strawberry-shake

**Reply by seangwright** [reply]
Awesome. I knew improvements were coming with v13, so I'll try that out when I get back to the project on Monday.
